injection/sharedmain/main_test.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sharedmain
+
+import (
+	"testing"
+
+	"knative.dev/pkg/injection"
+)
+
+func TestEnabledControllers(t *testing.T) {
+	tests := []struct {
+		name                string
+		disabledControllers []string
+		ctors               []injection.NamedControllerConstructor
+		want                int
+	}{
+		{
+			name:                "zero",
+			disabledControllers: []string{"foo"},
+			ctors:               []injection.NamedControllerConstructor{{Name: "bar"}},
+			want:                1,
+		},
+		{
+			name:                "one",
+			disabledControllers: []string{"foo"},
+			ctors:               []injection.NamedControllerConstructor{{Name: "foo"}},
+			want:                0,
+		},
+		{
+			name:                "two",
+			disabledControllers: []string{"foo"},
+			ctors: []injection.NamedControllerConstructor{
+				{Name: "foo"},
+				{Name: "bar"},
+			},
+			want: 1,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
```

Comment:
super nit, but do you need this if you don't call t.Parallel? (Maybe you want to add a `t.Parallel()` inside the run?)